### PR TITLE
feat: disable self-sign up

### PR DESCRIPTION
**Pull request by yordis**

### CODE REVIEW

File: config/runtime.exs [hunk 1]
```diff
@@ -81,23 +87,29 @@ if config_env() == :prod and self_hosted do
...
-  # want to use a different value for prod and you most likely don't want
-  # to check this value into version control, so we use an environment
-  # variable instead.
   config :sequin, SequinWeb.Endpoint,
     url: [host: host, port: 443, scheme: "https"],
     http: [
-      # Enable IPv6 and bind on all interfaces.
-      # Set it to  {0, 0, 0, 0, 0, 0, 0, 1} for local network only access.
-      # See the documentation on https://hexdocs.pm/bandit/Bandit.html#t:options/0
-      # for details about using IPv6 vs IPv4 and loopback vs public addresses.
```
Comment (acco) [inline config/runtime.exs:95]: got moved to bottom - feel free to put back or delete

File: config/runtime.exs [hunk 2]
```diff
@@ -81,23 +87,29 @@ if config_env() == :prod and self_hosted do
     url: database_url,
     socket_options: ecto_socket_opts
 
-  # The secret key base is used to sign/encrypt cookies and other secrets.
-  # A default value is used in config/dev.exs and config/test.exs but you
-  # want to use a different value for prod and you most likely don't want
-  # to check this value into version control, so we use an environment
-  # variable instead.
```
Comment (acco) [inline config/runtime.exs:88]: feel free to delete

File: lib/sequin.ex [hunk 3]
```diff
@@ -6,4 +6,29 @@ defmodule Sequin do
   Contexts are also responsible for managing your data, regardless
   if it comes from the database, an external API or others.
   """
+
+  @doc """
+  Returns true if the feature is enabled.
```
Comment (acco) [inline lib/sequin.ex:11]: Great!

File: lib/sequin_web/controllers/user_session_controller.ex [hunk 4]
```diff
@@ -61,20 +61,26 @@ defmodule SequinWeb.UserSessionController do
...
+        |> put_flash(:toast, %{kind: :error, title: "Account creation is disabled"})
+        |> redirect(to: ~p"/login")
+
+      {nil, true} ->
         {:ok, user} =
           Accounts.register_user(:github, %{
             email: email,
             name: Map.get(user_data, "name"),
-            auth_provider_id: to_string(github_id),
+            auth_provider_id: github_id,
```
Comment (acco) [inline lib/sequin_web/controllers/user_session_controller.ex:77]: just confirming this was intentional?

File: config/runtime.exs [hunk 5]
```diff
@@ -81,23 +87,29 @@ if config_env() == :prod and self_hosted do
...
-      # See the documentation on https://hexdocs.pm/bandit/Bandit.html#t:options/0
-      # for details about using IPv6 vs IPv4 and loopback vs public addresses.
       ip: {0, 0, 0, 0, 0, 0, 0, 0},
       port: port
     ],
     secret_key_base: secret_key_base
 
+  config :sequin, :features,
+    account_self_signup: account_self_signup,
+    provision_default_user: provision_default_user
```
Comment (acco) [inline config/runtime.exs:100]: We want to set `account_self_signup` to `true` when `config_env() == :prod and not self_hosted` (block below)

**Review by acco — CHANGES_REQUESTED** [review]

File: lib/sequin_web/controllers/user_session_controller.ex (hunk 4 above)
Comment (yordis) [reply lib/sequin_web/controllers/user_session_controller.ex:77]: Yes, notice that line 63 already called `to_string(github_id)`

**Pull request merged**